src/app/shared/services/programs/program.service.ts

```diff
@@ -44,19 +45,30 @@ export class ProgramService extends GenericListService {
     return program.pipe(
       switchMap((data) => {
         programWithDetail = data;
-        return  zip(
-          this.departmentService.getDepartment(data.department),
-          this.programCategoryService.getProgramCategory(data.category)
-        );
+        return  this.departmentService.getDepartment(data.department);
       }),
       map((data) => {
-        programWithDetail.department = data[0];
-        programWithDetail.category = data[1];
+        programWithDetail.department = data;
         return programWithDetail;
       }),
     );
   }
 
+  getProgramCategories() {
+    if (this.cachedProgramCategories === undefined) {
```

Comment:
```typescript
getProgramCategories() {
  if (this.cachedProgramCategories.length !== 0) {
    return observableOf(this.cachedProgramCategories);
  }
  // otherwise, do other stuff
}
```